### ISSUE TITLE
Add easier configuration for pygw Java gateway

### ISSUE THREAD

**Issue by jdgarrett** [locationtech/geowave#1791]
Right now it can be a bit confusing to set up `pygw` with custom Java gateways, such as the one provided by pyspark.  To alleviate this, we should add a separate gateway configuration module that allows the user to either provide a gateway object or gateway IP address and port.  This would provide a method of configuring `pygw` prior to import.

**Comment by jdgarrett** [comment]
Also, include documentation on using `pygw` with pyspark!

**Comment by jdgarrett** [comment]
This is addressed by #1798

**Comment by jdgarrett** [comment]
Once this is merged, you will be able to set the gateway prior to importing other `pygw` classes by doing the following:
```python
# This snippet should only be called once at the beginning of the application lifecycle
from pygw.gateway import gateway_config
gateway_config.set_gateway(sparkContext._gateway)

# Continue to use pygw normally
```
